algod/build/install.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+# Script to install algod in all sorts of different ways.
+#
+# Parameters:
+#    -d    : Location where binaries will be installed.
+#    -c    : Channel to install. Mutually exclusive with source options.
+#    -u    : Git repository URL. Mutually exclusive with -c.
+#    -b    : Git branch. Mutually exclusive with -c.
+#    -s    : (optional) Git Commit SHA hash. Mutually exclusive with -c.
+
+set -e
+
+rootdir=$(dirname "$0")
+pushd "$rootdir"
+
+BINDIR=""
+CHANNEL=""
+URL=""
+BRANCH=""
+SHA=""
```

Comment:
should these values just come from the environment compared to the command line arguments?

Reply:
I guess this is part and part docker inexperience.

By forcing all parameters to be provided to the script you can avoid some surprises from a maintenance perspective. i.e. a script doing something you didn't expect it to because you add a sensible default to what appeared to be an unused environment variable.

From the docker side, I'm never quite clear about where the build args are going to be available vs env variables.